pkg/controller/datavolume-controller.go

```diff
@@ -2779,6 +2851,39 @@ func getDeltaTTL(dv *cdiv1.DataVolume, ttl int32) time.Duration {
 	return delta
 }
 
+// handleCloneWithoutSource does error handling and validation of clones without source PVC
+func (r *DatavolumeReconciler) handleCloneWithoutSource(datavolume *cdiv1.DataVolume) (bool, error) {
+	sourcePvc, err := r.findSourcePvc(datavolume)
+	if err != nil {
+		// Clone without source
+		if k8serrors.IsNotFound(err) {
+			datavolume.Annotations[AnnNoSourceClone] = "true"
```

Comment:
Would it not make more sense to instead of setting this annotation, that we update the Running condition to false with `sourcePVC not found`? I am trying to understand the purpose of this annotation, and it is not actually used for anything, we create it and then remove it, but what is its purpose?

Reply:
When we create a clone without source, the clone-specific validation in the admission webhook is skipped, so my idea is to eventually check it once the PVC is created. I use that annotation to specify that a clone has been created without source so, when the source PVC is created, we do the validation. Once the validation succeeds, I remove the annotation because we just want to check the validation once, as we do in the webhook.

Maybe is not ideal to just use an annotation as a flag? Or it's better to rename it as something like `AnnCloneNeedsValidation`? I still see the point of having it unless we want to skip the validation and let the clone-controller handle it.

Reply:
I am saying why are you not always doing the validation if you see the PVC is there? Regardless of annotation.

Reply:
Yes, exactly, there is no reason that after we removed the annotation something didn't go wrong, and we need to reconcile, we still need to validate at that point, and with the annotation gone we won't. The validation is not very compute intensive, so why not just do it all the time. That way we will never miss it when we should have run it.

Reply:
hmmm I guess that, In that case, I can remove the annotation and just do the validation on every clone. However, wouldn't it be odd to keep repeating that validation on all clones independently of if they are created with or without source?